src/azure-cli/azure/cli/command_modules/vm/_vm_utils.py

```diff
@@ -541,6 +567,51 @@ def raise_unsupported_error_for_flex_vmss(vmss, error_message):
         raise ArgumentUsageError(error_message)
 
 
+def get_hyper_v_generation_from_vmss(cli_ctx, image_ref, location):  # pylint: disable=too-many-return-statements
```

Comment:
May I ask why do you want to put this method into the `_vm_utils.py`?

Reply:
> May I ask why do you want to put this method into the `_vm_utils.py`?

I thought this might be a common usage to get the `hyper_v_generation` from `ImageReference`. The method can be moved to `_validators.py` or `custom.py` as well.